frontends/ast/genrtlil.cc

```diff
@@ -717,6 +717,10 @@ struct AST_INTERNAL::ProcessGenerator
 						polarity.bits.push_back(RTLIL::S0);
 					}
 				}
+				if (polarity.bits.empty()) {
+					// TRG_POLARITY must be at least one bit, otherwise the "verilog" backend outputs .TRG_POLARITY({0{1'b0}})
```

Comment:
Is that a problem? I think the cell might fail check when reading it.

Reply:
tests/simple/string_format.v fails, since iverilog doesn't like the `{0{1'b0}}` part.

Reply:
Actually, I have a different question first: when does the output `.TRG_POLARITY({0{1'b0}})` ever happen? That sounds like a bug in and of itself.

Reply:
As far as I understand `tests/tools/autotest.sh`, it is generated by the `verilog` backend. This is the output in `tests/simple/string_format.out/string_format_syn0.v` without the change above:
```
/* Generated by Yosys 0.36+79 (git sha1 97d655a20, gcc 13.2.1 -fPIC -Os) */

module string_format_top();
  \$print  #(
    .ARGS_WIDTH(32'd168),
    .FORMAT("A: {168:> c}\n"),
    .PRIORITY(32'd4294967295),
    .TRG_ENABLE(32'd0),
    .TRG_POLARITY({0{1'b0}}),
    .TRG_WIDTH(32'd0)
  ) _0_ (
    .ARGS(168'h736f6d657468696e6720696e746572657374696e67),
    .EN(1'h1),
    .TRG()
  );
  \$print  #(
    .ARGS_WIDTH(32'd168),
    .FORMAT("B: {168:> c}\n"),
    .PRIORITY(32'd4294967294),
    .TRG_ENABLE(32'd0),
    .TRG_POLARITY({0{1'b0}}),
    .TRG_WIDTH(32'd0)
  ) _1_ (
    .ARGS(168'h736f6d657468696e6720696e746572657374696e67),
    .EN(1'h1),
    .TRG()
  );
endmodule
```

Reply:
Okay, so that's a bug in the Verilog backend where it does not output `$print` cells in `initial` blocks correctly. It should be fixed; any time the Verilog backend outputs a bare `$print` cell (without being explicitly requested to) is a bug in the backend.